core/notifications/src/history/repo.rs

```diff
@@ -59,11 +59,36 @@ impl PersistentNotifications {
         notification: &mut StatefulNotification,
     ) -> Result<(), NotificationHistoryError> {
         let mut tx = self.pool.begin().await?;
+        sqlx::query!(
+            r#"UPDATE stateful_notifications
+            SET acknowledged = true
```

Comment:
how do you know this is acknowledged = true? Need to inject the setting from the entity.